### PR TITLE
Integrate license check with PBC

### PR DESCRIPTION
**Pull request by pszczesniak**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Internal(ui): Integration of license check with PBC.

Closes https://github.com/cksource/ckeditor5-internal/issues/3226.
Closes https://github.com/cksource/ckeditor5-internal/issues/3249

---

### Additional information

_For example – encountered issues, assumptions you had to make, other affected tickets, etc._

### CODE REVIEW

File: packages/ckeditor5-ui/src/editorui/poweredby.ts [hunk 1]
```diff
@@ -94,6 +95,12 @@ export default class PoweredBy extends DomEmitterMixin() {
 	private _handleEditorReady(): void {
 		const editor = this.editor;
 
+		const licenseKey = this.editor.config.get( 'licenseKey' );
```
Comment (mlewand) [inline packages/ckeditor5-ui/src/editorui/poweredby.ts:98]:
```suggestion
		const licenseKey = editor.config.get( 'licenseKey' );
```

It's preassigned just a line above :D

File: packages/ckeditor5-ui/src/editorui/poweredby.ts [hunk 2]
```diff
@@ -94,6 +95,12 @@ export default class PoweredBy extends DomEmitterMixin() {
...
+
+		if ( licenseKey && verify( licenseKey ) === 'VALID' ) {
```
Comment (mlewand) [inline packages/ckeditor5-ui/src/editorui/poweredby.ts:100]: How about simplifying it down to:

```
if ( verify( editor.config.get( 'licenseKey' ) ) === 'VALID' ) {
```

?

The verify method must gracefully handle undefineds, nulls, falses etc.

**Review by mlewand — CHANGES_REQUESTED** [review]
Few minor remarks

File: packages/ckeditor5-utils/src/index.ts [hunk 3]
```diff
@@ -87,6 +87,7 @@ export { default as insertToPriorityArray } from './inserttopriorityarray';
 export { default as spliceArray } from './splicearray';
 
 export { default as uid } from './uid';
+export { default as verify } from './madewith';
```
Comment (Reinmar) [inline packages/ckeditor5-utils/src/index.ts:90]: Please use a better name. What does it verify?

File: packages/ckeditor5-ui/tests/manual/poweredby/poweredby.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # Powered by rendered inside of the editing root.
 
 1. Make sure the "powered by" link renders above the bottom edge of the editing root when focused.
-1. Make sure the link opens the page in a new tab.
-1. Make sure the link disappears as soon as the editor gets blurred.
-1. Make sure the link disappears when the editor is cropped by an ancestor with `overflow`.
+2. Make sure the link opens the page in a new tab.
```
[the remark this answers is not in the document]
Comment (pszczesniak) [reply packages/ckeditor5-ui/tests/manual/poweredby/poweredby.md]: Fixed in c2a0554

File: packages/ckeditor5-ui/src/editorui/poweredby.ts (hunk 2 above)
Comment (pszczesniak) [reply packages/ckeditor5-ui/src/editorui/poweredby.ts]: Done in 3af13cc

File: packages/ckeditor5-utils/src/index.ts (hunk 3 above)
Comment (pszczesniak) [reply packages/ckeditor5-utils/src/index.ts]: Fixed.

**Pull request merged**